### PR TITLE
[CELEBORN-891] Remove pipeline feature for sort based writer

### PR DESCRIPTION
**Pull request by pan3793**
### What changes were proposed in this pull request?

Remove pipeline feature for sort based writer

### Why are the changes needed?

The pipeline feature is added as part of CELEBORN-295, for performance. Eventually, an unresolvable issue that would crash the JVM was identified in https://github.com/apache/incubator-celeborn/pull/1807, and after discussion, we decided to delete this feature.

### Does this PR introduce _any_ user-facing change?

No, the pipeline feature is disabled by default, there are no changes to users who use the default settings.

### How was this patch tested?

Pass GA.

### CODE REVIEW

File: client-spark/spark-2/src/test/scala/org/apache/spark/shuffle/celeborn/CelebornShuffleManagerSuite.scala [hunk 1]
```diff
@@ -38,7 +38,6 @@ class SparkShuffleManagerSuite extends Logging {
       .set(s"spark.${CelebornConf.MASTER_ENDPOINTS.key}", "localhost:9097")
       .set(s"spark.${CelebornConf.CLIENT_PUSH_REPLICATE_ENABLED.key}", "false")
       .set("spark.shuffle.service.enabled", "false")
-      .set("spark.shuffle.useOldFetchProtocol", "true")
```
Comment (pan3793) [inline client-spark/spark-2/src/test/scala/org/apache/spark/shuffle/celeborn/CelebornShuffleManagerSuite.scala:41]: the configuration is added in Spark 3.0, should not be used in 2.x

File: client-spark/spark-3/src/main/java/org/apache/spark/shuffle/celeborn/SortBasedShuffleWriter.java [hunk 2]
```diff
@@ -189,28 +156,13 @@ public SortBasedShuffleWriter(
         conf,
         client,
         metrics,
-        executorService,
         sendBufferPool);
   }
 
   private void updatePeakMemoryUsed() {
-    // sorter can be null if this writer is closed
```
Comment (pan3793) [inline client-spark/spark-3/src/main/java/org/apache/spark/shuffle/celeborn/SortBasedShuffleWriter.java:197]: is the comment correct?

File: client-spark/spark-3/src/main/java/org/apache/spark/shuffle/celeborn/SortBasedShuffleWriter.java [hunk 3]
```diff
@@ -189,28 +156,13 @@ public SortBasedShuffleWriter(
...
-      for (SortBasedPusher pusher : pushers) {
-        if (pusher != null) {
-          long mem = pusher.getPeakMemoryUsedBytes();
-          if (mem > peakMemoryUsedBytes) {
-            peakMemoryUsedBytes = mem;
-          }
-        }
-      }
-    } else {
-      if (currentPusher != null) {
```
Comment (pan3793) [inline client-spark/spark-3/src/main/java/org/apache/spark/shuffle/celeborn/SortBasedShuffleWriter.java:208]: I don't find the null assignment code, so I think such checking is redundant

Comment (waitinfuture) [reply client-spark/spark-3/src/main/java/org/apache/spark/shuffle/celeborn/SortBasedShuffleWriter.java:208]: I think it's safe to remove this null check.

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks! Merging to main(v0.5.0)

**Comment by pan3793** [comment]
As this feature is broken, I cherry-picked it 0.4 to avoid confusing users.

**Pull request closed without merging**